AudioKit/Common/Internals/AudioKit+StartStop.swift

```diff
@@ -130,12 +130,18 @@ extension AudioKit {
                 // Note: doesn't throw since this is called from a notification observer
             }
         }
+        if Thread.isMainThread {
```

Comment:
@warpling If the notification. was called on the main thread you would get a deadlock on DispatchQueue.main.sync.  In general, use async unless there's a concrete reason to use sync on the main thread.

Reply:
Not really, no.

I'm not sure why it's there in the first place, but it was `DispatchQueue.main.sync` which was deadlocking on me.  So this just fixes the deadlock without having to read up why there was a dispatch_sync here originally.

Reply:
I suppose I’m just assuming that sync was a mistake.

Reply:
there's a number of examples like this one online (LIKE https://gist.github.com/sgr-ksmt/4880c5df5aeec9e558622cd6d5b477cb), with the isMainThread and sync on main, but doing a little more reading I found that it is not in fact safe as you found. I would assume that whoever wrote that saw this sort of safe dispatch example thinking it was exactly that. Seems like it's best to just get rid of the if statement entirely. I don't see why sync should be called on the main queue ever?

Reply:
I’d let @warpling answer this.  The fix is using async, not sync, so it should be good to go AFAIK.